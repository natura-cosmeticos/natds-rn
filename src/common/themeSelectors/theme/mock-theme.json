{
  "avatarSizes": {
    "tiny": {
      "size": 24,
      "fontSize": ".625rem"
    },
    "small": {
      "size": 32,
      "fontSize": ".875rem"
    },
    "standard": {
      "size": 40,
      "fontSize": "1rem"
    },
    "large": {
      "size": 48,
      "fontSize": "1rem"
    },
    "huge": {
      "size": 80,
      "fontSize": "1.5rem"
    }
  },
  "buttonSizes": {
    "small": {
      "paddingTop": 8,
      "paddingRight": 8,
      "paddingBottom": 8,
      "paddingLeft": 8,
      "minHeight": 32
    },
    "medium": {
      "paddingTop": 12,
      "paddingRight": 16,
      "paddingBottom": 12,
      "paddingLeft": 16,
      "minHeight": 40
    },
    "large": {
      "paddingTop": 16,
      "paddingRight": 16,
      "paddingBottom": 16,
      "paddingLeft": 16,
      "minHeight": 48
    }
  },
  "fontSize": {
    "h1": {
      "fontSize": "6rem",
      "fontWeight": 300
    },
    "h2": {
      "fontSize": "3.75rem",
      "fontWeight": 300
    },
    "h3": {
      "fontSize": "3rem",
      "fontWeight": 400
    },
    "h4": {
      "fontSize": "2.125rem",
      "fontWeight": 400
    },
    "h5": {
      "fontSize": "1.5rem",
      "fontWeight": 400
    },
    "h6": {
      "fontSize": "1.25rem",
      "fontWeight": 600
    },
    "subtitle1": {
      "fontSize": "1rem",
      "fontWeight": 400
    },
    "subtitle2": {
      "fontSize": "0.875rem",
      "fontWeight": 500
    },
    "body1": {
      "fontSize": "1rem",
      "fontWeight": 400
    },
    "body2": {
      "fontSize": "0.875rem",
      "fontWeight": 400
    },
    "button": {
      "fontSize": "0.875rem",
      "fontWeight": 600
    },
    "caption": {
      "fontSize": "0.75rem",
      "fontWeight": 400
    },
    "overline": {
      "fontSize": "0.75rem",
      "fontWeight": 400
    }
  },
  "iconSizes": {
    "micro": 16,
    "tiny": 24,
    "small": 32,
    "standard": 48,
    "medium": 64,
    "large": 72,
    "largex": 96,
    "largexx": 128,
    "huge": 144,
    "hugex": 192,
    "hugexx": 256
  },
  "opacity": {
    "12": 0.12,
    "16": 0.16,
    "24": 0.24,
    "32": 0.32,
    "48": 0.48,
    "56": 0.56,
    "64": 0.64,
    "80": 0.8,
    "00": 0,
    "04": 0.04,
    "08": 0.08
  },
  "overlay": {
    "solidBlack": 0.56
  },
  "radius": {
    "none": 0,
    "circle": "50%",
    "small": 2,
    "medium": 4,
    "large": 8
  },
  "sizes": {
    "none": 0,
    "micro": 4,
    "tiny": 8,
    "small": 16,
    "standard": 24,
    "semi": 32,
    "semix": 40,
    "medium": 48,
    "mediumx": 56,
    "large": 64,
    "largex": 72,
    "largexx": 80,
    "largexxx": 88,
    "huge": 96,
    "hugex": 128,
    "hugexx": 144,
    "hugexxx": 192,
    "veryhuge": 256
  },
  "spacing": {
    "spacing": 8,
    "spacingNone": 0,
    "spacingMicro": 4,
    "spacingTiny": 8,
    "spacingSmall": 16,
    "spacingStandard": 24,
    "spacingSemi": 32,
    "spacingLarge": 48,
    "spacingHuge": 64
  },
  "colorTokens": {
    "colorPrimary": "#F4AB34",
    "colorOnPrimary": "#333333",
    "colorPrimaryLight": "#FEFDE8",
    "colorOnPrimaryLight": "#333333",
    "colorPrimaryDark": "#EF8426",
    "colorOnPrimaryDark": "#333333",
    "colorSecondary": "#FF6B0B",
    "colorOnSecondary": "#000000",
    "colorSecondaryLight": "#FFF8E1",
    "colorOnSecondaryLight": "#000000",
    "colorSecondaryDark": "#FF5808",
    "colorOnSecondaryDark": "#000000",
    "colorBackground": "#FAFAFA",
    "colorOnBackground": "#333333",
    "colorSurface": "#FFFFFF",
    "colorOnSurface": "#333333",
    "colorHighlight": "#000000",
    "colorHighEmphasis": "#333333",
    "colorMediumEmphasis": "#777777",
    "colorLowEmphasis": "#BBBBBB",
    "colorLink": "#227BBD",
    "colorOnLink": "#FFFFFF",
    "colorSuccess": "#569A32",
    "colorOnSuccess": "#FFFFFF",
    "colorWarning": "#FCC433",
    "colorOnWarning": "#333333",
    "colorAlert": "#E74627",
    "colorOnAlert": "#FFFFFF"
  },
  "elevation": {
    "1": {
      "elevation": 1,
      "shadowColor": "#000000",
      "shadowOffset": { "height": 1, "width": 0 },
      "shadowOpacity": 0.14,
      "shadowRadius": 1
    },
    "2": {
      "elevation": 2,
      "shadowColor": "#000000",
      "shadowOffset": { "height": 2, "width": 0 },
      "shadowOpacity": 0.14,
      "shadowRadius": 2
    },
    "3": {
      "elevation": 3,
      "shadowColor": "#000000",
      "shadowOffset": { "height": 3, "width": 0 },
      "shadowOpacity": 0.14,
      "shadowRadius": 4
    },
    "4": {
      "elevation": 4,
      "shadowColor": "#000000",
      "shadowOffset": { "height": 4, "width": 0 },
      "shadowOpacity": 0.14,
      "shadowRadius": 5
    },
    "5": {
      "elevation": 5,
      "shadowColor": "#000000",
      "shadowOffset": { "height": 6, "width": 0 },
      "shadowOpacity": 0.14,
      "shadowRadius": 10
    },
    "6": {
      "elevation": 6,
      "shadowColor": "#000000",
      "shadowOffset": { "height": 8, "width": 0 },
      "shadowOpacity": 0.14,
      "shadowRadius": 10
    },
    "7": {
      "elevation": 7,
      "shadowColor": "#000000",
      "shadowOffset": { "height": 9, "width": 0 },
      "shadowOpacity": 0.14,
      "shadowRadius": 12
    },
    "8": {
      "elevation": 8,
      "shadowColor": "#000000",
      "shadowOffset": { "height": 12, "width": 0 },
      "shadowOpacity": 0.14,
      "shadowRadius": 17
    },
    "9": {
      "elevation": 9,
      "shadowColor": "#000000",
      "shadowOffset": { "height": 16, "width": 0 },
      "shadowOpacity": 0.14,
      "shadowRadius": 24
    },
    "10": {
      "elevation": 10,
      "shadowColor": "#000000",
      "shadowOffset": { "height": 24, "width": 0 },
      "shadowOpacity": 0.14,
      "shadowRadius": 38
    }
  }
}
