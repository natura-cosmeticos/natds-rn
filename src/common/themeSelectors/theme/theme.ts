import { tokens as styleTokens } from '@naturacosmeticos/natds-styles';
import { IColorThemeTokens } from '@naturacosmeticos/natds-styles/dist/tokens/colors/themeBound/IColorThemeTokens';
import { IElevation } from '@naturacosmeticos/natds-styles/dist/tokens/elevation/IElevation';

type Tokens = typeof styleTokens;
type TokensWithoutColor = Omit<Tokens, 'colors'>
type TokensWithoutColorTokens = Omit<TokensWithoutColor, 'colorTokens'>
type TokensWithoutElevation = Omit<TokensWithoutColorTokens, 'elevation'>

interface ElevationProps {
  elevation: number
  shadowColor: string
  shadowOffset: { height: number, width: number }
  shadowOpacity: number
  shadowRadius: number
}

type Elevation = Record<keyof IElevation, ElevationProps>


export type Theme = TokensWithoutElevation
  & {
    colorTokens: IColorThemeTokens,
    elevation: Elevation | {}
  }

type ThemeMode = 'dark' | 'light';

const parseElevation = (elevation: string) => {
  const regEx = /(\d+(\.\d+)?)/g;
  const matches = elevation.match(regEx);

  if (matches) {
    const [width, height, blurRadius, , r, g, b, alpha] = matches;

    const toUnit = (prop: string) => parseInt(prop, 0);

    return {
      elevation: 2,
      shadowColor: `rgb(${r}, ${g}, ${b})`,
      shadowOffset: { height: toUnit(height), width: toUnit(width) },
      shadowOpacity: parseFloat(alpha),
      shadowRadius: toUnit(blurRadius),
    };
  }

  return {};
};

const buidElevation = (elevation: IElevation) => {
  const nativeElevation = {};

  Object
    .keys(elevation)
    .forEach((key) => { nativeElevation[key] = parseElevation(elevation[key]); });

  return nativeElevation;
};

export const buildTheme = (tokens: Tokens, brand: string, mode: ThemeMode) => {
  const colorThemeName = `${brand}${mode === 'light' ? 'Light' : 'Dark'}ColorTokens`;
  const localTokens = { ...tokens };

  delete localTokens.colors;

  const theme: Theme = {
    ...localTokens,
    colorTokens: localTokens.colorTokens[colorThemeName],
    elevation: buidElevation(localTokens.elevation),
  };

  return theme;
};

interface ThemeSelector {
  (theme: Theme): string | undefined
}

export const getTheme = (selector: ThemeSelector) => (
  { theme }: { theme: Theme },
) => selector(theme);
