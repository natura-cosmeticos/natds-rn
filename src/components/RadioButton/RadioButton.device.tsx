import { storiesOf } from '@storybook/react-native';
import {
  all,
  interactive,
} from './RadioButton.stories';

storiesOf('RadioButton', module)
  .add('all', all)
  .add('interactive', interactive);
