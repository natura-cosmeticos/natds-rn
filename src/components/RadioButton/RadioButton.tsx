import React from 'react';

import { TouchableRipple } from '../TouchableRipple/TouchableRipple';
import {
  Container, Circle, Center, Label, Wrapper,
} from './RadioButton.styles';

export type RadioButtonColors = 'primary' | 'secondary';

export interface RadioButtonProps {
  /**
   * RadioButton color: `primary` | `secondary`
   */
  color?: RadioButtonColors;
  /**
   * Called when the touch is released, but not if cancelled
   */
  onPress?: (value: string) => void;
  /**
   * Programmatically tell if the component is selected
   */
  selected?: boolean;
  /**
   * Optional property that tells if the select is disabled,
   * if so, the user cannot click on this item.
   */
  disabled?: boolean;
  /**
   * Optional label that appears to the right of the component,
   * if the user clicks that label, the onPress function will be called
   */
  label?: string;
  /**
   * The value of the component.
   */
  value?: string;
  /**
   * Optional testID
   */
  testID?: string;
}

export const RadioButton = ({
  color = 'primary',
  onPress,
  selected,
  disabled = false,
  label,
  value = '',
  testID = 'radio-button',
}: RadioButtonProps) => {
  const onPressWithValue = () => {
    if (onPress) {
      onPress(value);
    }
  };
  const radio = (
    <TouchableRipple
      size={20}
      color={color}
      onPress={onPressWithValue}
      disabled={disabled}
      testID={testID}
      style={{ borderRadius: 50 }}
    >
      <Wrapper>
        <Circle disabled={disabled} color={color} selected={selected} testID={`${testID}-circle`}>
          {
            selected
              ? (
                <Center
                  disabled={disabled}
                  color={color}
                  selected={selected}
                  testID={`${testID}-circle-selected`}
                />
              ) : null
          }
        </Circle>
      </Wrapper>
    </TouchableRipple>
  );

  if (label) {
    return (
      <Container disabled={disabled} testID={`${testID}-label`} onPress={onPressWithValue}>
        {radio}
        <Label disabled={disabled}>{label}</Label>
      </Container>
    );
  }

  return radio;
};
