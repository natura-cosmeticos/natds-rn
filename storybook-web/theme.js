/* eslint-disable sort-keys */
import { create } from '@storybook/theming/create';
import themes from '@naturacosmeticos/natds-themes/react-native';

const buildTheme = (theme, mode) => create({
  base: mode,
  colorPrimary: theme.color.primary,
  colorSecondary: theme.color.secondary,

  // UI
  appBg: theme.color.surface,
  appContentBg: theme.color.background,

  // Text
  textColor: theme.color.highEmphasis,
  textInverseColor: theme.color.lowEmphasis,

  // Toolbar
  barBg: theme.color.surface,
  barTextColor: theme.color.highEmphasis,

  // Brand
  brandTitle: 'Natura Design System',
});

export const dark = buildTheme(themes.natura.dark, 'dark');
export const light = buildTheme(themes.natura.light, 'light');
