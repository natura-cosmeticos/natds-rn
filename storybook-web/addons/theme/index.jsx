import React, { useEffect, Fragment } from 'react';
import { useAddonState, Consumer } from '@storybook/api';
import { TabButton, WithTooltip, TooltipLinkList } from '@storybook/components';
import {
  PANEL_ID, CHANGE, PARAM_KEY, getStoryBookTheme,
} from './shared';
import { light, dark } from '../../theme';

import './styles.css';

const DEFAULT_THEME = {
  name: 'natura',
  type: 'light',
};

const buildThemeProps = mode => ({ ...DEFAULT_THEME, type: mode });

function parseTheme(themes) {
  return Object.entries(themes).reduce((accum, [name, variants]) => {
    Object.entries(variants).forEach(([type]) => {
      accum.push({ name, type });
    });

    return accum;
  }, []);
}

const createBackgroundSelectorItem = ({ name, type }, onChange, selectedItem) => ({
  active: selectedItem.name === name && selectedItem.type === type,
  id: `${name} - ${type}`,
  onClick: () => onChange({ name, type }),
  title: `${name} - ${type}`,
});

const getDisplayedItems = (list, onChange, selectedItem) => {
  if (!list.length) {
    return [];
  }

  return [
    ...list.map(params => createBackgroundSelectorItem(params, onChange, selectedItem)),
  ];
};

const mapper = ({ api, state }) => {
  const story = state.storiesHash[state.storyId];
  const { themes = {} } = (story && api.getParameters(story.id, PARAM_KEY)) || {};

  return { items: parseTheme(themes) };
};

export default function Theme({ channel, api }) {
  const apiTheme = getStoryBookTheme();
  const initialTheme = apiTheme ? buildThemeProps(apiTheme) : DEFAULT_THEME;
  const [currentTheme, changeTheme] = useAddonState(PANEL_ID, initialTheme);
  const { disabled } = api.getCurrentParameter(PARAM_KEY) || {};

  const handleChange = (params) => {
    changeTheme(params);
    api.setOptions({
      theme: params.type === 'light' ? light : dark,
    });
    const event = new Event('CHANGE_THEME', { detail: params.type });

    window.dispatchEvent(event);
    channel.emit(CHANGE, params);
  };

  const renderTooltip = items => ({ onHide }) => (
    <TooltipLinkList
      links={getDisplayedItems(items, (index) => {
        handleChange(index);
        onHide();
      }, currentTheme)}
    />
  );

  const eventListener = event => handleChange(buildThemeProps(event.matches ? 'dark' : 'light'));

  useEffect(() => {
    handleChange(initialTheme);

    const matchMediaEnvent = window.matchMedia('(prefers-color-scheme: dark)');

    matchMediaEnvent.addEventListener('change', eventListener);

    return () => {
      matchMediaEnvent.removeEventListener('change', eventListener);
    };
  }, []);

  if (disabled) return null;

  return (
    <Consumer filter={mapper}>
      {({ items }) => (items.length ? (
          <Fragment>
            <WithTooltip
              placement="top"
              trigger="click"
              closeOnClick
              tooltip={renderTooltip(items)}
            >
              <TabButton
                key="Theme"
                title="Change the Theme">
                {currentTheme.name} - {currentTheme.type}
              </TabButton>
            </WithTooltip>
          </Fragment>
      ) : null)}
    </Consumer>
  );
}
