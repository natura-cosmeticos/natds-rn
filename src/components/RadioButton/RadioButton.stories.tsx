import React, { useState } from 'react';
import { View } from 'react-native';
import { boolean, select, text } from '@storybook/addon-knobs';

import { RadioButton, RadioButtonColors } from './RadioButton';

export default {
  component: RadioButton,
  parameters: {
    componentSubtitle: 'Selection controls allow the user to select options.',
  },
  title: 'Components|RadioButton',
};

const colorTypes = {
  primary: 'primary',
  secondary: 'secondary',
};

export const all = () => {
  const [selected, setSelected] = useState<string | null>(null);

  return (
    <View style={{ maxWidth: 600, padding: 30 }}>
      <RadioButton
        color="primary"
        selected={selected === '1'}
        onPress={setSelected}
        label="Primary"
        value='1'
      />
      <RadioButton
        color="secondary"
        selected={selected === '2'}
        onPress={setSelected}
        label="Secondary"
        value='2'
      />
      <RadioButton
        label="Disabled"
        disabled
      />
      <RadioButton
        selected={true}
        label="Selected disabled"
        disabled
      />
    </View>
  );
};


export const interactive = () => (
  <View style={{ maxWidth: 600, padding: 30 }}>
    <RadioButton
      color={select('Color', colorTypes, 'primary') as RadioButtonColors}
      selected={boolean('Selected', true)}
      disabled={boolean('Disabled', false)}
      onPress={() => null}
      label={text('Label', 'My Label')}
      value={text('Value', 'my-label')}
    />
  </View>
);
