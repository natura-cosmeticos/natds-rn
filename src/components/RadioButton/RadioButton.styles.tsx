
import styled from 'styled-components/native';
import {
  getColorPrimary,
  Theme,
  getColorSecondary,
  getColorLowEmphasis,
  getColorOnBackground,
  getColorMediumEmphasis,
} from '../../common/themeSelectors';
import { RadioButtonColors } from './RadioButton';

function getCircleColor(
  selected = false,
  disabled: boolean,
  color: RadioButtonColors,
  theme: Theme,
) {
  if (disabled) {
    return getColorLowEmphasis(theme);
  }

  if (selected) {
    return color === 'secondary' ? getColorSecondary(theme) : getColorPrimary(theme);
  }

  return getColorMediumEmphasis(theme);
}

export const Container = styled.TouchableOpacity.attrs({
  activeOpacity: 1,
})`
  flex-direction: row;
  align-items: center;
`;

export const Label = styled.Text<{ theme: Theme; disabled: boolean }>(({ theme, disabled }) => ({
  color: disabled ? getColorLowEmphasis(theme) : getColorOnBackground(theme),
  fontSize: 14,
  letterSpacing: 0.22,
  marginLeft: theme.spacing.spacingTiny,
}));


export const Circle = styled.View<{
  theme: Theme;
  color: RadioButtonColors;
  selected?: boolean;
  disabled: boolean;
}>(({
  theme,
  color,
  selected,
  disabled,
}) => ({
  alignItems: 'center',
  borderColor: getCircleColor(selected, disabled, color, theme),
  borderRadius: 12,
  borderWidth: '2px',
  height: 20,
  justifyContent: 'center',
  width: 20,
}));

export const Center = styled.View<{
  theme: Theme;
  color: RadioButtonColors;
  selected?: boolean;
  disabled: boolean;
}>(({
  theme,
  color,
  selected,
  disabled,
}) => ({
  backgroundColor: getCircleColor(selected, disabled, color, theme),
  borderRadius: 12,
  height: 10,
  width: 10,
}));
