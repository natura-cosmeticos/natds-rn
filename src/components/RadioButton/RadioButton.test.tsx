import React from 'react'
import { ThemeProvider } from 'styled-components/native'
import { fireEvent, render } from '@testing-library/react-native'
import renderer from 'react-test-renderer'
import theme from '../../common/themeSelectors/theme/mock-theme.json'
import { RadioButton, RadioButtonProps } from './RadioButton'

jest.mock(
  'react-native/Libraries/Components/Touchable/TouchableOpacity',
  () => 'TouchableOpacity'
)

const renderRadioButton = (fn, props?: Omit<RadioButtonProps, 'theme'>) => (fn(
  <ThemeProvider theme={theme}>
    <RadioButton {...props} />
  </ThemeProvider>
))

describe('RadioButton component', () => {
  it('should render radio button correctly', () => {
    const radioButton = renderRadioButton(renderer.create).toJSON()

    expect(radioButton).toMatchSnapshot()
  })

  it('should render radio button selected correctly', () => {
    const radioButton = renderRadioButton(renderer.create, { selected: true }).toJSON()

    expect(radioButton).toMatchSnapshot()
  })

  it('should render radio button with label correctly', () => {
    const radioButton = renderRadioButton(renderer.create, { label: 'My Label' }).toJSON()

    expect(radioButton).toMatchSnapshot()
  })

  it('should render radio button selected but disabled correctly', () => {
    const radioButton = renderRadioButton(renderer.create, { disabled: true }).toJSON()

    expect(radioButton).toMatchSnapshot()
  })

  it('should render radio button selected but disabled with label correctly', () => {
    const radioButton = renderRadioButton(renderer.create, { disabled: true, label: 'My Label' }).toJSON()

    expect(radioButton).toMatchSnapshot()
  })

  it('should call the onPress function when the user touches the radio button', () => {
    const onPress = jest.fn()
    const { queryByTestId } = renderRadioButton(render, {
      onPress,
      value: 'radio-button'
    })

    const radioButton = queryByTestId('radio-button')

    fireEvent.press(radioButton)

    expect(onPress).toHaveBeenCalledWith('radio-button')
  })

  it('should call the onPress function when the user touches the radio button label', () => {
    const onPress = jest.fn()
    const { queryByTestId } = renderRadioButton(render, {
      label: 'My label',
      onPress,
      value: 'radio-button-label'
    })

    const label = queryByTestId('radio-button-label')

    fireEvent.press(label)

    expect(onPress).toHaveBeenCalledWith('radio-button-label')
  })

  it('should render circle inside radio when selected is true', () => {
    const onPress = jest.fn()
    const { queryByTestId } = renderRadioButton(render, {
      onPress,
      selected: true,
      value: 'radio-button'
    })

    const radioButtonCircle = queryByTestId('radio-button-circle-selected')

    expect(radioButtonCircle).not.toBeNull()
  })

  it('should render radio inside label container when label is different from null/undefined', () => {
    const onPress = jest.fn()
    const { queryByTestId } = renderRadioButton(render, {
      label: 'My Label',
      onPress,
      value: 'radio-button'
    })

    const radioButtonLabel = queryByTestId('radio-button-label')

    expect(radioButtonLabel).not.toBeNull()
  })
})
