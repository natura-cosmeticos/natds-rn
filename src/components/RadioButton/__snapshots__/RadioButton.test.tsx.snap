// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioButton component should render radio button correctly 1`] = `
<TouchableWithoutFeedback
  delayPressOut={255}
  disabled={false}
  onPress={[Function]}
  onPressIn={[Function]}
  onPressOut={[Function]}
  testID="radio-button"
>
  <View
    size={40}
    style={
      Array [
        Object {
          "alignItems": "center",
          "height": 40,
          "justifyContent": "center",
          "width": 40,
        },
      ]
    }
  >
    <View
      color="primary"
      size={40}
      style={
        Object {
          "backgroundColor": "#F4AB34",
          "borderRadius": 20,
          "height": 40,
          "left": 0,
          "opacity": 0,
          "position": "absolute",
          "top": 0,
          "transform": Array [
            Object {
              "scale": 0,
            },
          ],
          "width": 40,
        }
      }
    />
    <View>
      <View
        color="primary"
        disabled={false}
        style={
          Array [
            Object {
              "alignItems": "center",
              "borderColor": "#777777",
              "borderRadius": 12,
              "borderWidth": 2,
              "height": 20,
              "justifyContent": "center",
              "width": 20,
            },
          ]
        }
        testID="radio-button-circle"
      />
    </View>
  </View>
</TouchableWithoutFeedback>
`;

exports[`RadioButton component should render radio button selected but disabled correctly 1`] = `
<TouchableWithoutFeedback
  delayPressOut={255}
  disabled={true}
  onPress={[Function]}
  onPressIn={[Function]}
  onPressOut={[Function]}
  testID="radio-button"
>
  <View
    size={40}
    style={
      Array [
        Object {
          "alignItems": "center",
          "height": 40,
          "justifyContent": "center",
          "width": 40,
        },
      ]
    }
  >
    <View
      color="primary"
      size={40}
      style={
        Object {
          "backgroundColor": "#F4AB34",
          "borderRadius": 20,
          "height": 40,
          "left": 0,
          "opacity": 0,
          "position": "absolute",
          "top": 0,
          "transform": Array [
            Object {
              "scale": 0,
            },
          ],
          "width": 40,
        }
      }
    />
    <View>
      <View
        color="primary"
        disabled={true}
        style={
          Array [
            Object {
              "alignItems": "center",
              "borderColor": "#BBBBBB",
              "borderRadius": 12,
              "borderWidth": 2,
              "height": 20,
              "justifyContent": "center",
              "width": 20,
            },
          ]
        }
        testID="radio-button-circle"
      />
    </View>
  </View>
</TouchableWithoutFeedback>
`;

exports[`RadioButton component should render radio button selected but disabled with label correctly 1`] = `
<TouchableOpacity
  activeOpacity={1}
  disabled={true}
  onPress={[Function]}
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexDirection": "row",
      },
    ]
  }
  testID="radio-button-label"
>
  <TouchableWithoutFeedback
    delayPressOut={255}
    disabled={true}
    onPress={[Function]}
    onPressIn={[Function]}
    onPressOut={[Function]}
    testID="radio-button"
  >
    <View
      size={40}
      style={
        Array [
          Object {
            "alignItems": "center",
            "height": 40,
            "justifyContent": "center",
            "width": 40,
          },
        ]
      }
    >
      <View
        color="primary"
        size={40}
        style={
          Object {
            "backgroundColor": "#F4AB34",
            "borderRadius": 20,
            "height": 40,
            "left": 0,
            "opacity": 0,
            "position": "absolute",
            "top": 0,
            "transform": Array [
              Object {
                "scale": 0,
              },
            ],
            "width": 40,
          }
        }
      />
      <View>
        <View
          color="primary"
          disabled={true}
          style={
            Array [
              Object {
                "alignItems": "center",
                "borderColor": "#BBBBBB",
                "borderRadius": 12,
                "borderWidth": 2,
                "height": 20,
                "justifyContent": "center",
                "width": 20,
              },
            ]
          }
          testID="radio-button-circle"
        />
      </View>
    </View>
  </TouchableWithoutFeedback>
  <Text
    disabled={true}
    style={
      Array [
        Object {
          "color": "#BBBBBB",
          "fontSize": 14,
          "letterSpacing": 0.22,
          "marginLeft": 8,
        },
      ]
    }
  >
    My Label
  </Text>
</TouchableOpacity>
`;

exports[`RadioButton component should render radio button selected correctly 1`] = `
<TouchableWithoutFeedback
  delayPressOut={255}
  disabled={false}
  onPress={[Function]}
  onPressIn={[Function]}
  onPressOut={[Function]}
  testID="radio-button"
>
  <View
    size={40}
    style={
      Array [
        Object {
          "alignItems": "center",
          "height": 40,
          "justifyContent": "center",
          "width": 40,
        },
      ]
    }
  >
    <View
      color="primary"
      size={40}
      style={
        Object {
          "backgroundColor": "#F4AB34",
          "borderRadius": 20,
          "height": 40,
          "left": 0,
          "opacity": 0,
          "position": "absolute",
          "top": 0,
          "transform": Array [
            Object {
              "scale": 0,
            },
          ],
          "width": 40,
        }
      }
    />
    <View>
      <View
        color="primary"
        disabled={false}
        selected={true}
        style={
          Array [
            Object {
              "alignItems": "center",
              "borderColor": "#F4AB34",
              "borderRadius": 12,
              "borderWidth": 2,
              "height": 20,
              "justifyContent": "center",
              "width": 20,
            },
          ]
        }
        testID="radio-button-circle"
      >
        <View
          color="primary"
          disabled={false}
          selected={true}
          style={
            Array [
              Object {
                "backgroundColor": "#F4AB34",
                "borderRadius": 12,
                "height": 10,
                "width": 10,
              },
            ]
          }
        />
      </View>
    </View>
  </View>
</TouchableWithoutFeedback>
`;

exports[`RadioButton component should render radio button with label correctly 1`] = `
<TouchableOpacity
  activeOpacity={1}
  disabled={false}
  onPress={[Function]}
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexDirection": "row",
      },
    ]
  }
  testID="radio-button-label"
>
  <TouchableWithoutFeedback
    delayPressOut={255}
    disabled={false}
    onPress={[Function]}
    onPressIn={[Function]}
    onPressOut={[Function]}
    testID="radio-button"
  >
    <View
      size={40}
      style={
        Array [
          Object {
            "alignItems": "center",
            "height": 40,
            "justifyContent": "center",
            "width": 40,
          },
        ]
      }
    >
      <View
        color="primary"
        size={40}
        style={
          Object {
            "backgroundColor": "#F4AB34",
            "borderRadius": 20,
            "height": 40,
            "left": 0,
            "opacity": 0,
            "position": "absolute",
            "top": 0,
            "transform": Array [
              Object {
                "scale": 0,
              },
            ],
            "width": 40,
          }
        }
      />
      <View>
        <View
          color="primary"
          disabled={false}
          style={
            Array [
              Object {
                "alignItems": "center",
                "borderColor": "#777777",
                "borderRadius": 12,
                "borderWidth": 2,
                "height": 20,
                "justifyContent": "center",
                "width": 20,
              },
            ]
          }
          testID="radio-button-circle"
        />
      </View>
    </View>
  </TouchableWithoutFeedback>
  <Text
    disabled={false}
    style={
      Array [
        Object {
          "color": "#333333",
          "fontSize": 14,
          "letterSpacing": 0.22,
          "marginLeft": 8,
        },
      ]
    }
  >
    My Label
  </Text>
</TouchableOpacity>
`;
